src/parser/statement.js

```diff
@@ -303,35 +303,34 @@ pp.parseForStatement = function (node) {
     return this.parseFor(node, init);
   }
 
+  if (this.hasPlugin("lightscript") && (!forAwait)) {
+    // idx, elem, key, or val begins a LS-enhanced loop.
+    if (
+      (this.isContextual("idx") || this.isContextual("elem") || this.isContextual("key") || this.isContextual("val"))
+    ) {
+      return this.parseEnhancedForIn(node);
+    }
+  }
+
   const refShorthandDefaultPos = { start: 0 };
   const init = this.parseExpression(true, refShorthandDefaultPos);
   if (this.match(tt._in) || this.isContextual("of")) {
+    if (this.hasPlugin("lightscript") && (!init.isNowAssign)) {
+      this.raise(this.state.lastTokStart, "For-in requires `const` or `now`.");
```

Comment:
whoops, my earlier comment about the error message may also have been swallowed... let me dig that up...